### PR TITLE
Added Item discription.

### PR DESCRIPTION
**Pull request by Mahnoor-Faisal0303**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+
 const App: React.FC = () => {
   return (
-    <div>
-      <CustomTable data={data} headers={headers} />
-    </div>
+    <Router>
+      <Routes>
+        <Route path="/" element={<CustomTable data={data} headers={headers} />} />
+        <Route path="/details/:id" element={<DetailComponent />} />
```
Comment (saad277) [inline src/App.tsx]: make a constant folder and in that 
makes a routes.ts file 
then make 
enum APP_ROUTES{
DETAILS_PAGE="/details/:id"
}

Comment (saad277) [reply src/App.tsx]: we will map a route with constants so incase if a route name changes we dont have to go and change everywhere just the constant

File: src/Components/CustomTable.tsx [hunk 2]
```diff
@@ -53,6 +56,7 @@ const CustomTable: React.FC<TableProps> = ({ headers }) => {
                                 {Object.keys(Data).map((key) => (
                                     <TableCell key={key}>{Data[key]}</TableCell>
                                 ))}
+                                <ButtonStyle onClick={() => navigate('details/' + Data.id)}>Description</ButtonStyle>
```
Comment (saad277) [inline src/Components/CustomTable.tsx]: discuss this on slack

File: src/Data/Data.ts [hunk 3]
```diff
@@ -1,101 +1,101 @@
...
-        name: "Raspberry ripple cheesecake",
-        calories: 320,
-        fat: 17.0,
-        carbs: 38,
-        protein: 4.2
-    },
-    { id: 49, name: "Caramel apple pie", calories: 290, fat: 14.5, carbs: 35, protein: 3.6 },
-    { id: 50, name: "Honey lavender gelato", calories: 170, fat: 8.0, carbs: 22, protein: 2.0 }
-];
+// export const data: Item[] = [
```
Comment (saad277) [inline src/Data/Data.ts]: remove commented code

File: src/Components/DetailTable.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { detailData } from '../Data/Detail'
+import { useParams } from 'react-router-dom';
+import { TypographyStyle, TypographyStyle2 } from './Style';
+export const DetailComponent: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const itemId = +id!;  //coverted nullable string into non-nullable number
```
Comment (saad277) [inline src/Components/DetailTable.tsx]: why is this needed /

File: src/Data/Data.ts (hunk 3 above)
Comment (Mahnoor-Faisal0303) [reply src/Data/Data.ts]: ok

File: src/Components/DetailTable.tsx (hunk 4 above)
Comment (Mahnoor-Faisal0303) [reply src/Components/DetailTable.tsx]: by the use of useParams it returns string but we need a number so i converted this, and id shouldn't null ever, that'swhy usse "!" this ,,

File: src/App.tsx (hunk 1 above)
Comment (Mahnoor-Faisal0303) [reply src/App.tsx]: ok
